File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let result = 0;
    let basket = [];

    for (let i = 0; i < moves.length; i++) {
        let column = moves[i] - 1;
        for (let j = 0; j < board.length; j++) {
            if (board[j][column] !== 0) {
                let doll = board[j][column];
                board[j][column] = 0; // 인형 제거
                if (basket.length > 0 && basket[basket.length - 1] === doll) {
                    basket.pop();
                    result += 2; // 터트려진 인형 개수 증가
                } else {
                    basket.push(doll);
                }
                break; // 다음 크레인 작동으로 이동
            }
        }
    }

    return result;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    N = len(board)
    out = []
    count = 0
    for move in moves:
        for i in range(N):
            if board[i][move-1] != 0:
                count += 1
                out.append(board[i][move-1])
                board[i][move-1] = 0
                break
    flag = True
    while flag:
        flag = False
        for i in range(len(out)-1):
            if out[i] == out[i+1]:
                answer += 2
                out.pop(i)
                out.pop(i)
                flag = True
                break
    return answer